nucliadb_node/src/replication/replicator.rs

```diff
@@ -360,15 +370,21 @@ pub async fn connect_to_primary_and_replicate_forever(
     settings: Arc<Settings>,
     shard_cache: Arc<AsyncUnboundedShardWriterCache>,
     secondary_id: String,
+    shutdown_notified: Arc<AtomicBool>,
 ) -> NodeResult<()> {
     loop {
+        if shutdown_notified.load(std::sync::atomic::Ordering::Relaxed) {
+            return Ok(());
+        }
         let result = connect_to_primary_and_replicate(
-            settings.clone(),
-            shard_cache.clone(),
+            Arc::clone(&settings),
+            Arc::clone(&shard_cache),
             secondary_id.clone(),
+            Arc::clone(&shutdown_notified),
         )
         .await;
-        if result.is_err() {
+
+        if let Err(_) = result {
             error!(
                 "Error happened during replication. Will retry: {:?}",
                 result
```

Comment:
Consider the following to get rid of the clippy warning: 
```rust
        if result.is_ok() {
            return Ok(());
        }

        error!(
            "Error happened during replication. Will retry: {:?}",
            result
        );
        tokio::time::sleep(std::time::Duration::from_secs(
            settings.replication_delay_seconds(),
        ))
        .await;
```

Reply:
it's funny, I was trying to be more rustic!